### PR TITLE
feat(codebuild): add support for setting the gitCloneDepth property on Project sources

### LINKED ISSUE #1789: Allow to specify GitCloneDepth in codebuild.Project
While defining the source in `codebuild.Project` resource I noticed [GitCloneDepth](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codebuild-project-source.html#cfn-codebuild-project-source-gitclonedepth) property is missing. Hopefully, it is available in the `codebuild.CfnProject`.

### PR DESCRIPTION
**Pull request by skinny85**
Also fixed the weird casing of the 'GitHubEnterPrise' SourceType enum value while I was in the area.

Fixes #1789

----

### Pull Request Checklist

* [x] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [x] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
* [ ] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [x] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-codebuild/lib/source.ts [hunk 1]
```diff
@@ -71,17 +71,48 @@ export class NoSource extends BuildSource {
...
-export interface CodeCommitSourceProps extends BuildSourceProps {
+export interface CodeCommitSourceProps extends GitBuildSourceProps {
   repository: codecommit.IRepository;
 }
 
 /**
  * CodeCommit Source definition for a CodeBuild project.
  */
-export class CodeCommitSource extends BuildSource {
+export class CodeCommitSource extends GitBuildSource {
```
Comment (eladb) [inline packages/@aws-cdk/aws-codebuild/lib/source.ts]: BTW, based on our discussions around #1743, this class should technically be in CodeCommit...

File: packages/@aws-cdk/aws-codebuild/lib/source.ts [hunk 2]
```diff
@@ -71,17 +71,48 @@ export class NoSource extends BuildSource {
...
+   * If this value is 0, greater than 25, or not provided,
+   * then the full history is downloaded with each build of the project.
+   */
+  cloneDepth?: number;
+}
+
+/**
+ * A common superclass of all build sources that are backed by Git.
+ */
+export abstract class GitBuildSource extends BuildSource {
```
Comment (eladb) [inline packages/@aws-cdk/aws-codebuild/lib/source.ts]: I think this is an example of over-modeling. Now what happens if we have another Git source that's not in the codebuild package, then what...

File: packages/@aws-cdk/aws-codebuild/lib/source.ts [hunk 3]
```diff
@@ -318,7 +349,7 @@ export enum SourceType {
   CodeCommit = 'CODECOMMIT',
   CodePipeline = 'CODEPIPELINE',
   GitHub = 'GITHUB',
-  GitHubEnterPrise = 'GITHUB_ENTERPRISE',
```
Comment (eladb) [inline packages/@aws-cdk/aws-codebuild/lib/source.ts]: LOLx

File: packages/@aws-cdk/aws-codebuild/lib/source.ts (hunk 1 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-codebuild/lib/source.ts]: I think you're probably right, but that seems like a separate change.

File: packages/@aws-cdk/aws-codebuild/lib/source.ts (hunk 2 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-codebuild/lib/source.ts]: But I was so proud of how I DRYied it up... 😭

What alternative would you propose here Elad?

Comment (eladb) [reply packages/@aws-cdk/aws-codebuild/lib/source.ts]: Just duplicate. Sometimes it's too DRY

Comment (eladb) [reply packages/@aws-cdk/aws-codebuild/lib/source.ts]: But I am okay with this too :-D

Comment (skinny85) [reply packages/@aws-cdk/aws-codebuild/lib/source.ts]: I can't. I can't bear myself to copy & paste `cloneDepth` after I've so beautifully applied inheritance and the Decorator pattern here. My object-oriented programming education forbids me.

I'll leave this as is for now, feel free to open a separate PR changing this behind my back 😃

**Pull request merged**